test/integration/targets/supervisorctl/files/sendProcessStdin.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+import signal
+import sys
+import time
+
+proc = sys.argv[1]
+value = sys.argv[2]
+logpath = sys.argv[3]
+username = sys.argv[4]
+password = sys.argv[5]
+
+if sys.version_info[0] == 2:
+    from xmlrpclib import ServerProxy
+    from urllib import quote
+else:
+    from xmlrpc.client import ServerProxy
+    from urllib.parse import quote
+
+if username:
+    url = 'http://%s:%s@127.0.0.1:9001/RPC2' % (quote(username, safe=''), quote(password, safe=''))
+else:
+    url = 'http://127.0.0.1:9001/RPC2'
+
+server = ServerProxy(url, verbose=True)
+server.supervisor.sendProcessStdin(proc, 'import sys; print(%s); sys.stdout.flush();\n' % value)
+
+pid = int(server.supervisor.getPID())
+# supervisord will close and reopen the main activity log and all child log files.
+os.kill(pid, signal.SIGUSR2)
+
+with open(logpath) as log:
```

Comment:
It looks like there's a race condition here. The log file could be read before supervisord has reopened it.

Reply:
Updated: supervisord is now stopped before reading log file.